### DISCUSSION TITLE
How to write a reaction that is independent of humidity?

### DISCUSSION THREAD

**Discussion by mst-parkhachev** [firemodels/fds#13569] [General]
Dear colleagues!

I have a question: how can I write a combustion reaction that would be correct at any air humidity?
Basically, in the manual you can find examples in which the air composition is specified (or the default composition is used). For example, like this:

```
&SPEC ID='NITROGEN', LUMPED_COMPONENT_ONLY=T /
&SPEC ID='OXYGEN', LUMPED_COMPONENT_ONLY=T /
&SPEC ID='WATER VAPOR', LUMPED_COMPONENT_ONLY=T /
&SPEC ID='CARBON DIOXIDE', LUMPED_COMPONENT_ONLY=T /

&SPEC ID='AIR', BACKGROUND=T,
SPEC_ID(1)='OXYGEN', VOLUME_FRACTION(1)=0.20756,
SPEC_ID(2)='NITROGEN', VOLUME_FRACTION(2)=0.78257,
SPEC_ID(3)='WATER VAPOR', VOLUME_FRACTION(3)=0.00948,
SPEC_ID(4)='CARBON DIOXIDE', VOLUME_FRACTION(4)=0.00039 /
```

Then the name of the fuel, the composition of the reaction products, and the reaction itself are specified:
```
&SPEC ID='METHANE' /

&SPEC ID='PRODUCTS',
SPEC_ID(1)='CARBON DIOXIDE', VOLUME_FRACTION(1)=0.09438,
SPEC_ID(2)='WATER VAPOR', VOLUME_FRACTION(2)=0.19663,
SPEC_ID(3)='NITROGEN', VOLUME_FRACTION(3)=0.70899 /

&REAC FUEL='METHANE', SPEC_ID_NU='METHANE','AIR','PRODUCTS',
NU=-1,-9.6357,10.6357, HEAT_OF_COMBUSTION=50000. /
```

In this case, one of the reactants is the mixture – air. The components of "air" that do not participate in the reaction are transferred to the reaction products.

If you try to change the content of water vapor in the original "air" mixture, FDS will report that the reaction is not balanced. I need to perform calculations at different humidity, and it is inconvenient to calculate the reaction coefficients each time. In addition, nitrogen and water vapor are not reactants, so it seems that it would be logical to write the reaction something like this.

```
&SPEC ID='OXYGEN' /
&SPEC ID='FUEL_1'  MW = 16.04/

&SPEC ID='PRODUCTS',
SPEC_ID(1)='CARBON DIOXIDE', VOLUME_FRACTION(1) = 1,
SPEC_ID(2)='WATER VAPOR', VOLUME_FRACTION(2) = 2 /

&REAC FUEL='FUEL_1', SPEC_ID_NU='FUEL_1','OXYGEN','PRODUCTS',
NU=-1,-2,1, HEAT_OF_COMBUSTION=50000. /
```
[test_05.txt](https://github.com/user-attachments/files/17355070/test_05.txt)

Instead of explicitly specifying "methane", I used the name 'FUEL_1'. (Using the name "methane" had the same effect.)
The composition of the "air" mixture is not explicitly specified in this example. As far as I understand, FDS knows this mixture as the default background mixture.

If we take into account that water vapor will be released during combustion and the concentration of moisture in the air will change, then assigning the reaction to a form that does not depend on humidity seems even more relevant. Moreover, writing the reaction in a form that depends on humidity seems incorrect due to the change in humidity during combustion.

When simulating my reaction example, FDS does not report an error, but no combustion occurs either. The CHID_hrr.csv file shows non-zero MLR_FUEL_1 consumption, but MLR_OXYGEN and MLR_PRODUCTS are zero. So, no combustion occurs.

I suppose that FDS for some reason does not understand that oxygen should be taken from the "air".

Please tell me how I can record the initial data and reaction so that water vapor is not included in the initial reagents? At the same time, I would like to regulate the humidity with the HUMIDITY parameter, and not calculate the air composition manually for each humidity. It is also important that the humidity changing due to combustion does not affect the balance of the combustion reaction.

And one more question. If I measure humidity with a device, QUANTITY='RELATIVE HUMIDITY', will it take into account the increase in humidity due to the release of moisture during combustion, or not?

**Comment by mst-parkhachev** [comment]
Do I understand correctly that if the air composition is set explicitly, then the HUMIDITY parameter does not affect anything?

**Comment by drjfloyd** [comment]
Per the guide SPEC_ID_NU must be a tracked species. If you want to change humidty you could create a spreadsheet, python script, etc. to generate the inputs or just don't use a lumped species for AIR. Track N2, O2, CO2, etc.  independently then you could use O2 on te REAC line.

**Comment by mst-parkhachev** [comment]
Do the devices (QUANTITY='RELATIVE HUMIDITY' and 'VOLUME FRACTION') measure the total amount of water vapor in both spices - AIR and PRODUCTS? Looking at the function RELATIVE_HUMIDITY(Y_H2O,TEMP,PZONE) code, it seems to me that the device with QUANTITY='RELATIVE HUMIDITY' takes into account only water vapor in the AIR mixture. Is it true that to measure the humidity of the entire gas mixture in a room after combustion I need to use a device with QUANTITY= 'VOLUME FRACTION'?

Thanks for your replies, Dr. Floyd!

**Reply by drjfloyd** [reply]
When you reference a primitive species in an output, you get that species summed over any tracked species it is a part of.  So the relative humidity includes water vapor from air, products, and (if you had them) droplet evaporation.
